plugins/modules/database/influxdb/influxdb_retention_policy.py

```diff
@@ -272,30 +287,58 @@ def alter_retention_policy(module, client, retention_policy):
     module.exit_json(changed=changed)
 
 
+def drop_retention_policy(module, client):
+    database_name = module.params['database_name']
+    policy_name = module.params['policy_name']
+
+    changed = False
+
+    if not module.check_mode:
+        try:
+            client.drop_retention_policy(policy_name, database_name)
+        except exceptions.InfluxDBClientError as e:
+            module.fail_json(msg=e.content)
+        changed = True
+    module.exit_json(changed=changed)
+
+
 def main():
     argument_spec = InfluxDb.influxdb_argument_spec()
     argument_spec.update(
+        state=dict(default='present', type='str', choices=['present', 'absent']),
         database_name=dict(required=True, type='str'),
         policy_name=dict(required=True, type='str'),
-        duration=dict(required=True, type='str'),
-        replication=dict(required=True, type='int'),
+        duration=dict(required=False, type='str'),
+        replication=dict(required=False, type='int'),
         default=dict(default=False, type='bool'),
         shard_group_duration=dict(required=False, type='str'),
     )
     module = AnsibleModule(
         argument_spec=argument_spec,
-        supports_check_mode=True
+        supports_check_mode=True,
+        required_if=(
+            ('state', 'present', ['duration', 'replication']),
+        )
```

Comment:
```suggestion
        ),
```
If you add this comma, the next time someone adds another kwarg this line does not need to be touched :)